examples/siem-connector/main.go

```diff
@@ -92,96 +280,91 @@ func (a *App) NewTenant(ctx context.Context, id string) error {
 	}
 
 	// Store the Tenant and associated Synqly objects in an in-memory cache.
-	a.Tenants = append(a.Tenants, &Tenant{
-		ID:              id,
-		SynqlyAccountId: account.Result.Account.Id,
-		SynqlyClient:    client,
-		EventLogger:     nil,
+	app.Tenants = append(app.Tenants, &Tenant{
+		ID:                 id,
+		SynqlyAccountId:    account.Result.Account.Id,
+		SynqlyClient:       client,
+		SynqlyEngineClient: nil,
 	})
 	return nil
 }
 
-// configureEventLogging initializes event logging for a Tenant.
-// This example uses Splunk SIEM providers and mock SIEM provider as the event logging
-// providers; however, an Integration can be configured to target any supported provider type.
-// Stores the HEC_TOKEN as a secure Credential object, then creates a Splunk Integration
-// targeting SPLUNK_URL.
-//
-// Returns an error if the Tenant cannot be found, or if an Integration cannot
-// be created for the given Tenant.
-func (a *App) configureEventLogging(ctx context.Context, tenantID, siemProviderType string) error {
-	// Find the tenant
-	var tenant *Tenant
-	for _, t := range a.Tenants {
-		if t.ID == tenantID {
-			tenant = t
-			break
-		}
-	}
-	if tenant == nil {
-		return fmt.Errorf("tenant not found")
+// query performs a query against the SIEM provider for each tenant.
+func (app *App) query(ctx context.Context) error {
+	req := &engine.QuerySiemEventsRequest{
+		Cursor:         nil,
+		IncludeRawData: engine.Bool(true),
+		Order:          []*string{engine.String("time[asc]")},
+		Limit:          engine.Int(10),
 	}
 
-	// Configure an Integration in Synqly depending on this tenant's config
-	var providerConfig *mgmt.ProviderConfig
-	switch siemProviderType {
-	case "splunk":
-		// We need to save the tenant's Splunk credentials in Synqly before configuring the Integration
-		// We will use the Synqly Client we created for the tenant to do this
-		credential, err := tenant.SynqlyClient.Credentials.Create(ctx, tenant.SynqlyAccountId, &mgmt.CreateCredentialRequest{
-			Fullname: mgmt.String(fmt.Sprintf("%s authentication token", siemProviderType)),
-			Config: &mgmt.CredentialConfig{
-				Token: &mgmt.TokenCredential{
-					Secret: splunkToken,
-				},
-			},
-		})
+	for _, tenant := range app.Tenants {
+		res, err := boff.RetryWithData(func() (*engine.QuerySiemEventsResponse, error) {
```

Comment:
I do like the use of backoff here but you can do it in a simpler way -- our SDK has built in retries with backoff. You would do

```
	client := mgmtClient.NewClient(
		mgmtClient.WithMaxAttempts(5),
		mgmtClient.WithToken(xxx),
	)
```